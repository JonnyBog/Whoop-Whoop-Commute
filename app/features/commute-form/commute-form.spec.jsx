import React from 'react';
import rendererShallow from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

import CommuteForm from 'features/commute-form/commute-form';

describe('Features', () => {
  describe('Commute Form component', () => {
    it('should export a function', () => {
      expect(CommuteForm).toBeDefined();
    });

    let props;

    beforeEach(() => {
      props = {
        requestCommuteFormData: jest.fn(),
        data: [],
        isFetching: false,
        error: '',
        values: [],
        errors: [],
        handleChange: jest.fn(),
        handleSubmit: jest.fn(),
        setFieldValue: jest.fn(),
        submitCount: jest.fn(),
        touched: []
      };
    });

    afterEach(() => {
      props = null;
    });

    /**
     * createShallowRenderTree
     * @param {Object} updatedProps - defaults to props
     * @returns {JSX} shallow renderer component
     */
    function createShallowRenderTree (updatedProps = props) {
      const shallowRenderer = rendererShallow.createRenderer();

      return shallowRenderer.render(
        <CommuteForm {...updatedProps} />
      );
    }

    /**
     * createShallowWrapper
     * @param   {Object} updatedProps - defaults to props
     * @returns {JSX} shallow component
     */
    function createShallowWrapper (updatedProps = props) {
      return shallow(
        <CommuteForm {...updatedProps} />
      );
    }

    describe('Appearance', () => {
      it('should render', () => {
        const tree = createShallowRenderTree();
        expect(tree).toMatchSnapshot();
      });

      it('should show error when there is an error', () => {
        const updatedProps = Object.assign(
          {},
          props,
          {
            error: 'test error'
          }
        );
        const wrapper = createShallowWrapper(updatedProps);

        expect(wrapper.find('FormError')).toHaveLength(1);
      });
    });
  });
});
