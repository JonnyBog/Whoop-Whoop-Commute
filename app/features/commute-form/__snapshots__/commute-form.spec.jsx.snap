// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Commute Form component Appearance should render 1`] = `
<React.Fragment>
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "halfMileRadius": 1,
        "lat": 51.4272,
        "lng": -0.0543,
        "workStation": "",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": false,
        "_nodes": Array [
          "workStation",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "workStation": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  >
    [Function]
  </Formik>
  
  <CommuteFormResults
    data={Array []}
  />
</React.Fragment>
`;
